lib/libvfio-user.c

```diff
@@ -1006,13 +1034,21 @@ process_request(vfu_ctx_t *vfu_ctx)
         return ret;
     }
 
-    ret = exec_command(vfu_ctx, &hdr, ret, fds, &nr_fds, _iovecs, &iovecs,
+    ret = exec_command(vfu_ctx, &hdr, ret, fds, nr_fds, _iovecs, &iovecs,
                        &nr_iovecs, &free_iovec_data);
 
+    for (i = 0; i < nr_fds; i++) {
+        if (fds[i] != -1) {
+            vfu_log(vfu_ctx, VFU_INF, "closing fd=%d", fds[i]);
```

Comment:
sorry, I should have been a bit clearer: this is an error, but one causable by a malicious client. So it should be debug-only, but still clarify it's not normal, e.g.

vfu_log(vfu_ctx, VFU_DBG, "closing unexpected fd %d (index %zu) from cmd %u", fds[i], i, hdr.cmd);

Reply:
It's not necessarily caused by a malicious client. While _additional_ FDs are malicious and will be closed, if a failure occurs we will be closing a legitimate FD. Currently this can only happen if we run out of DMA regions.

Reply:
Another case where the FD is not unexpected is setting up interrupts where the client passed the wrong flags.

Reply:
That's true enough. We still probably want to survey at some point our logging for when it *can* be malicious though

Reply:
In this particular case can tell which FDs are legitimate and which aren't, I've created https://github.com/nutanix/libvfio-user/issues/141 for that.